### ISSUE TITLE
Scan failed with copyright-summary

### ISSUE THREAD

**Issue by pombredanne** [nexB/scancode-toolkit#1072]
Reported by @tdruez 

Using the code in the https://github.com/nexB/scancode-toolkit/tree/377-1043-summaries branch and scanning an older version of scancode itself, we have this crash:
```
$ wget https://github.com/nexB/scancode-toolkit/releases/download/v2.2.1/scancode-toolkit-2.2.1.tar.bz2
$ extractcode  .
Extracting archives...
[####################] 550                                                                  
Extracting done.

$ scancode -c --copyright-summary tmp/scancode/ --json-pp copysumbug.json -n3 --verbose
 Setup plugin: output:json-pp...
 Setup plugin: post_scan:copyright-summary...
 Setup plugin: scan:copyrights...
Collect file inventory...
Scan files for: copyrights with 3 process(es)...
[.....
.....]

Run post-scans...
 Run post-scan: copyright-summary...
ERROR: failed to run post-scan plugin: copyright-summary:
Traceback (most recent call last):
  File "/tmp/src/scancode/cli.py", line 801, in run_plugins
    plugin.process_codebase(codebase, **kwargs)
  File "/tmp/src/scancode/plugin_copyright_summary.py", line 151, in process_codebase
    summarized = summarize(summaries)
  File "/tmp/src/scancode/plugin_copyright_summary.py", line 244, in summarize
    clusters = list(cluster(texts))
  File "/tmp/src/scancode/plugin_copyright_summary.py", line 269, in cluster
    texts.sort(key=lambda x:-len(x.key))
  File "/tmp/src/scancode/plugin_copyright_summary.py", line 269, in <lambda>
    texts.sort(key=lambda x:-len(x.key))
TypeError: object of type 'NoneType' has no len()
```

And furthermore there is an empty JSON file created.

**Comment by tdruez** [comment]
Same issue running ScanCode on https://pypi.python.org/packages/8c/79/e739f2d86a1c2595b1b7b5202991cb6022b680c82d5ccde2f3783ff13adf/django_grappelli-2.11.1-py2.py3-none-any.whl#md5=e51751e1cc49eeb7e7aab4172d96fc28

Probably a better package to use for debugging as it's way smaller than ScanCode itself.

**Comment by pombredanne** [comment]
This has been fixed and merged in develop and now available in the latest tags 2.9.7 and up
